pkg/beacon/client/multi.go

```diff
@@ -0,0 +1,202 @@
+	for _, client := range b.Clients {
+		go client.SubscribeToHeadEvents(ctx, slotC)
+	}
+}
+
+func (b *MultiBeaconClient) GetProposerDuties(epoch structs.Epoch) (*RegisteredProposersResponse, error) {
+	// return the first successful beacon node response
+	clients := b.clientsByLastResponse()
+
+	for i, client := range clients {
+		log := b.Log.WithField("endpoint", client.Endpoint())
+
+		duties, err := client.GetProposerDuties(epoch)
+		if err != nil {
+			log.WithError(err).Error("failed to get proposer duties")
+			continue
+		}
+
+		b.bestBeaconIndex.Store(int64(i))
+
+		// Received successful response. Set this index as last successful beacon node
+		return duties, nil
+	}
+
+	return nil, ErrNodesUnavailable
+}
+
+func (b *MultiBeaconClient) SyncStatus() (*SyncStatusPayloadData, error) {
+	var bestSyncStatus *SyncStatusPayloadData
+	var foundSyncedNode bool
+
+	// Check each beacon-node sync status
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+	for _, instance := range b.Clients {
+		wg.Add(1)
+		go func(client BeaconNode) {
+			defer wg.Done()
+			log := b.Log.WithField("endpoint", client.Endpoint())
+
+			syncStatus, err := client.SyncStatus()
+			if err != nil {
+				log.WithError(err).Error("failed to get sync status")
+				return
+			}
+
+			mu.Lock()
+			defer mu.Unlock()
+
+			if foundSyncedNode {
+				return
+			}
+
+			if bestSyncStatus == nil {
+				bestSyncStatus = syncStatus
+			}
+
+			if !syncStatus.IsSyncing {
+				bestSyncStatus = syncStatus
+				foundSyncedNode = true
+			}
+		}(instance)
+	}
+
+	// Wait for all requests to complete...
+	wg.Wait()
+
+	if !foundSyncedNode {
+		return nil, ErrBeaconNodeSyncing
+	}
+
+	if bestSyncStatus == nil {
+		return nil, ErrNodesUnavailable
+	}
+
+	return bestSyncStatus, nil
+}
+
+func (b *MultiBeaconClient) KnownValidators(headSlot structs.Slot) (AllValidatorsResponse, error) {
+	// return the first successful beacon node response
+	clients := b.clientsByLastResponse()
+
+	for i, client := range clients {
+		log := b.Log.WithField("endpoint", client.Endpoint())
+
+		validators, err := client.KnownValidators(headSlot)
+		if err != nil {
+			log.WithError(err).Error("failed to fetch validators")
+			continue
+		}
+
+		b.bestBeaconIndex.Store(int64(i))
+
+		// Received successful response. Set this index as last successful beacon node
+		return validators, nil
+	}
+
+	return AllValidatorsResponse{}, ErrNodesUnavailable
+}
+
+func (b *MultiBeaconClient) Genesis() (genesisInfo structs.GenesisInfo, err error) {
+	clients := b.clientsByLastResponse()
+	for _, client := range clients {
+		if genesisInfo, err = client.Genesis(); err != nil {
+			b.Log.WithError(err).
+				WithField("endpoint", client.Endpoint()).
+				Warn("failed to get genesis info")
+			continue
+		}
+
+		return genesisInfo, nil
+	}
+
+	return genesisInfo, err
+}
+
+func (b *MultiBeaconClient) Endpoint() string {
+	return b.clientsByLastResponse()[0].Endpoint()
```

Comment:
So. Under some degree of misuse of this structure you'll get nilpointer here. 
because clientsByLastResponse would return b.Clients that can be nil. It's better safe than sorry :)